Source/Scene/ModelExperimental/ModelExperimentalSkin.js

```diff
@@ -0,0 +1,176 @@
+import Matrix4 from "../../Core/Matrix4.js";
+import Check from "../../Core/Check.js";
+import defaultValue from "../../Core/defaultValue.js";
+
+/**
+ * An in-memory representation of a skin that affects nodes in the {@link ModelExperimentalSceneGraph}.
+ * Skins should only be initialized after all of the {@link ModelExperimentalNode}s have been instantiated
+ * by the scene graph.
+ *
+ * @param {Object} options An object containing the following options:
+ * @param {ModelComponents.Skin} options.skin The corresponding skin components from the 3D model
+ * @param {ModelExperimentalSceneGraph} options.sceneGraph The scene graph this skin belongs to.
+ *
+ * @alias ModelExperimentalSkin
+ * @constructor
+ *
+ * @private
+ */
+export default function ModelExperimentalSkin(options) {
+  options = defaultValue(options, defaultValue.EMPTY_OBJECT);
+  //>>includeStart('debug', pragmas.debug);
+  Check.typeOf.object("options.skin", options.skin);
+  Check.typeOf.object("options.sceneGraph", options.sceneGraph);
+  //>>includeEnd('debug');
+
+  this._sceneGraph = options.sceneGraph;
+  const skin = options.skin;
+
+  this._skin = skin;
+
+  // Do we need this?
+  this._dirty = false;
+
+  this._inverseBindMatrices = undefined;
+  this._joints = [];
+  this._jointMatrices = [];
+
+  initialize(this);
+}
+
+Object.defineProperties(ModelExperimentalSkin.prototype, {
+  /**
+   * The internal skin this runtime skin represents.
+   *
+   * @memberof ModelExperimentalSkin.prototype
+   * @type {ModelComponents.Skin}
+   * @readonly
+   *
+   * @private
+   */
+  skin: {
+    get: function () {
+      return this._skin;
+    },
+  },
+
+  /**
+   * The scene graph this skin belongs to.
+   *
+   * @type {ModelExperimentalSceneGraph}
+   * @readonly
+   *
+   * @private
+   */
+  sceneGraph: {
+    get: function () {
+      return this._sceneGraph;
+    },
+  },
+
+  /**
+   * The inverse bind matrices of the skin.
+   *
+   * @memberof ModelExperimentalSkin.prototype
+   * @type {Matrix4[]}
+   * @readonly
+   *
+   * @private
+   */
+  inverseBindMatrices: {
+    get: function () {
+      return this._inverseBindMatrices;
+    },
+  },
+
+  /**
+   * The joints of the skin.
+   *
+   * @memberof ModelExperimentalSkin.prototype
+   * @type {ModelExperimentalNode[]}
+   * @readonly
+   *
+   * @private
+   */
+  joints: {
+    get: function () {
+      return this._joints;
+    },
+  },
+
+  /**
+   * The joint matrices for the skin, where each joint matrix is computed as
+   * [jointMatrix] = [jointWorldTransform] * [bindMatrix].
```

Comment:
wait, `bindMatrix` is really `inverseBindMatrix` based on the code below? 

I would use `inverseBindMatrix` everywhere to be unambiguous